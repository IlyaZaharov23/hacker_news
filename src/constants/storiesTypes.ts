export const STORIES_SHOWED_TYPE = {
  ASK: "ASK",
  BEST: "BEST",
  JOB: "JOB",
  NEW: "NEW",
  SHOW: "SHOW",
  TOP: "TOP",
};

