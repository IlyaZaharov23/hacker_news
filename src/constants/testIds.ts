export const TEST_ID = {
  START_PAGE: {
    START_PAGE_ROOT: "start-page-root",
    GO_TO_NEWS_BUTTON: "go-to-news-button",
    APP_LOGO: "app-logo",
    APP_TITLE: "app-title",
    SWITCHER_BUTTON: "switcher-button",
    ACTIONS_TITLE: "actions-title",
    FOOTER_QUOTE_SKELETON: "footer-quote-skeleton",
    FOOTER_QUOTE_TEXT: "footer-quote-text",
    FOOTER_ROOT: "footer-root",
    CUSTOM_DIVIDER: "custom-divider",
  },
  NEWS_PAGE: {
    NEWS_PAGE_ROOT: "news-page-root",
    NEWS_ITEM_IN_LIST: (id: number | undefined) => `news-list-in-list-${id}`,
  },
  STORY_PAGE: {
    STORY_PAGE_ROOT: (id: number | undefined) => `story-page-root-${id}`,
    STORY_HEADER: (id: number | undefined) => `story-header-${id}`,
    STORY_HEADER_SKELETON: `story-header-skeleton`,
    STORY_HEADER_TITLE: (id: number | undefined) => `story-header-title-${id}`,
    STORY_HEADER_AUTHOR: (id: number | undefined) =>
      `story-header-author-${id}`,
    STORY_HEADER_PUBLISHED: (id: number | undefined) =>
      `story-header-published-${id}`,
    STORY_HEADER_REDIRECT_TO_STORY: (id: number | undefined) =>
      `story-header-redirect-to-story-${id}`,
  },
  COMMENTS: {
    COMMENTS_ROOT: (id: number | undefined) => `comments-root-${id}`,
    COMMENT_ITEM: (id: number | undefined) => `comment-item-${id}`,
    SHOW_NESTED_COMMENS_BUTTON: (id: number | undefined) =>
      `show-nested-comments-button-${id}`,
    HIDE_NESTED_COMMENTS_BUTTON: (id: number | undefined) =>
      `hide-nested-comments-button-${id}`,
    NESTED_COMMENTS_ROOT: (id: number | undefined) =>
      `nested-comments-root-${id}`,
    COMMENTS_SKELETON: "comments-skeleton",
    COMMENTS_PLACEHOLDER: "comments-placeholder",
  },
  HEADER: {
    HEADER_ROOT: "header-root",
    BACK_BUTTON: "back-button",
    UPDATE_BUTTON: "update-button",
    MENU_BUTTON: "menu-button",
    APP_LOGO: "app-logo",
  },
  NEWS_SKELETON: "news-skeleton",
  NEWS_PLACEHOLDER: "news-placeholder",
  NEWS_TYPE_SWITCHER: {
    SWITCHER_ROOT: "switcher-root",
    SWITCHER_ITEM: (type: string) => `switcher-item-${type}`,
  },
};
