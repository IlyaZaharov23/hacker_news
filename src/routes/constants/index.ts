export const SCREEN_ROUTES = {
    START_PAGE: '/',
    NEWS_LIST: '/news',
    NEWS_ITEM: '/news/:id'
}