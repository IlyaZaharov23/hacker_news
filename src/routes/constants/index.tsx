import { lazy } from "react";

const NewsList = lazy(() => import("../../components/News"));
const StartPage = lazy(() => import("../../components/StartPage"));
const StoryPage = lazy(() => import("../../components/StoryPage"));

const SCREEN_NAMES = {
  START_PAGE: "START_PAGE",
  NEWS_LIST: "NEWS_LIST",
  NEWS_ITEMS: "NEWS_ITEM",
};

export const SCREEN_ROUTES = {
  START_PAGE: "/",
  NEWS_LIST: "/news",
  NEWS_ITEM: "/story/:id",
};

const routes = [
  {
    name: SCREEN_NAMES.START_PAGE,
    key: "start_page",
    path: SCREEN_ROUTES.START_PAGE,
    component: <StartPage />,
  },
  {
    name: SCREEN_NAMES.NEWS_LIST,
    key: "news_list",
    path: SCREEN_ROUTES.NEWS_LIST,
    component: <NewsList />,
  },
  {
    name: SCREEN_NAMES.NEWS_ITEMS,
    key: "news_item",
    path: SCREEN_ROUTES.NEWS_ITEM,
    component: <StoryPage />,
  },
];

export default routes;
