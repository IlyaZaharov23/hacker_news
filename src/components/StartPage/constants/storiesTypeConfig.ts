import { STORIES_SHOWED_TYPE } from "constants/storiesTypes";

export const storiesTypeConfig = {
  [STORIES_SHOWED_TYPE.ASK]: "Ask Stories",
  [STORIES_SHOWED_TYPE.BEST]: "Best Stories",
  [STORIES_SHOWED_TYPE.JOB]: "Job Stories",
  [STORIES_SHOWED_TYPE.NEW]: "New Stories",
  [STORIES_SHOWED_TYPE.SHOW]: "Show Stories",
  [STORIES_SHOWED_TYPE.TOP]: "Top Stories",
};
