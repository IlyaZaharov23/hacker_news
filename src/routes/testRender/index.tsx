import { Suspense } from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { RenderRoutes } from "routes/renderRoutes";
import { store } from "store";
import { ProgressLoader } from "components/ProgressLoader";

export const TestRender = () => {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <Suspense fallback={<ProgressLoader fullScreen />}>
          <RenderRoutes />
        </Suspense>
      </MemoryRouter>
    </Provider>
  );
};
