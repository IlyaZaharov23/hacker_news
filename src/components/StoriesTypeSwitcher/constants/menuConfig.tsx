import GradeIcon from "@mui/icons-material/Grade";
import FiberNewIcon from "@mui/icons-material/FiberNew";
import LiveHelpIcon from "@mui/icons-material/LiveHelp";
import WorkIcon from "@mui/icons-material/Work";
import PushPinIcon from "@mui/icons-material/PushPin";
import GitHubIcon from "@mui/icons-material/GitHub";
import { STORIES_SHOWED_TYPE } from "constants/storiesTypes";

export const menuItems = [
  {
    value: STORIES_SHOWED_TYPE.ASK,
    title: "Ask",
    icon: (sx: any) => <LiveHelpIcon sx={sx} />,
  },
  {
    value: STORIES_SHOWED_TYPE.BEST,
    title: "Best",
    icon: (sx: any) => <GradeIcon sx={sx} />,
  },
  {
    value: STORIES_SHOWED_TYPE.JOB,
    title: "Job",
    icon: (sx: any) => <WorkIcon sx={sx} />,
  },
  {
    value: STORIES_SHOWED_TYPE.NEW,
    title: "New",
    icon: (sx: any) => <FiberNewIcon sx={sx} />,
  },
  {
    value: STORIES_SHOWED_TYPE.SHOW,
    title: "Show",
    icon: (sx: any) => <GitHubIcon sx={sx} />,
  },
  {
    value: STORIES_SHOWED_TYPE.TOP,
    title: "Top",
    icon: (sx: any) => <PushPinIcon sx={sx} />,
  },
];
