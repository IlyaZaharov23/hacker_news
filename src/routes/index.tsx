import { lazy } from "react";
import { SCREEN_ROUTES } from "./constants";

const NewsList = lazy(() => import("../components/News"));
const StartPage = lazy(() => import("../components/StartPage"));

const SCREEN_NAMES = {
  START_PAGE: "START_PAGE",
  NEWS_LIST: "NEWS_LIST",
  NEWS_ITEMS: "NEWS_ITEM",
};

const routes = [
  {
    name: SCREEN_NAMES.START_PAGE,
    key: "start_page",
    path: SCREEN_ROUTES.START_PAGE,
    component: <StartPage />,
  },
  {
    name: SCREEN_NAMES.NEWS_LIST,
    key: "news_list",
    path: SCREEN_ROUTES.NEWS_LIST,
    component: <NewsList />,
  },
  {
    name: SCREEN_NAMES.NEWS_ITEMS,
    key: "news_item",
    path: SCREEN_ROUTES.NEWS_ITEM,
    component: "",
  },
];

export default routes;
