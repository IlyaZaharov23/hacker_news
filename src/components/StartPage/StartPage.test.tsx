import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { STORIES_SHOWED_TYPE } from "constants/storiesTypes";
import { TEST_ID } from "constants/testIds";
import { SCREEN_ROUTES } from "routes/constants";
import { TestRender } from "routes/testRender";

describe("START_PAGE_TEST", () => {
  test("page elements are rendered", async () => {
    render(<TestRender baseRoutes={[SCREEN_ROUTES.START_PAGE]} />);
    const pageRoot = await screen.findByTestId(
      TEST_ID.START_PAGE.START_PAGE_ROOT
    );
    const appLogo = screen.getByTestId(TEST_ID.START_PAGE.APP_LOGO);
    const goToNewsButton = screen.getByTestId(
      TEST_ID.START_PAGE.GO_TO_NEWS_BUTTON
    );
    const actionsTitle = screen.getByTestId(TEST_ID.START_PAGE.ACTIONS_TITLE);
    const storiesTypeSwitcher = screen.getByTestId(
      TEST_ID.START_PAGE.SWITCHER_BUTTON
    );
    const footerRoot = screen.getByTestId(TEST_ID.START_PAGE.FOOTER_ROOT);
    expect(pageRoot).toBeInTheDocument();
    expect(appLogo).toBeInTheDocument();
    expect(goToNewsButton).toBeInTheDocument();
    expect(actionsTitle).toBeInTheDocument();
    expect(storiesTypeSwitcher).toBeInTheDocument();
    expect(footerRoot).toBeInTheDocument();
  });
  test("Navigate to news", async () => {
    render(<TestRender baseRoutes={[SCREEN_ROUTES.START_PAGE]} />);
    const button = await screen.findByTestId(
      TEST_ID.START_PAGE.GO_TO_NEWS_BUTTON
    );
    userEvent.click(button);
    expect(
      await screen.findByTestId(TEST_ID.NEWS_PAGE.NEWS_PAGE_ROOT)
    ).toBeInTheDocument();
  });
  test("Switcher click", async () => {
    render(<TestRender baseRoutes={[SCREEN_ROUTES.START_PAGE]} />);
    const switcherButton = screen.getByTestId(
      TEST_ID.START_PAGE.SWITCHER_BUTTON
    );
    userEvent.click(switcherButton);
    expect(
      await screen.findByTestId(TEST_ID.NEWS_TYPE_SWITCHER.SWITCHER_ROOT)
    ).toBeInTheDocument();
    const topNewsItems = await screen.findByTestId(
      TEST_ID.NEWS_TYPE_SWITCHER.SWITCHER_ITEM(STORIES_SHOWED_TYPE.TOP)
    );
    userEvent.click(topNewsItems);
    expect(
      screen.queryByTestId(TEST_ID.NEWS_TYPE_SWITCHER.SWITCHER_ROOT)
    ).not.toBeInTheDocument();
  });
});
