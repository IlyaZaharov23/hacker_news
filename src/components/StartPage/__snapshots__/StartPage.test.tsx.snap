// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`START_PAGE_TEST matches snapshot 1`] = `[Function]`;
